The-SMIRNOFF-force-field-format.md

```diff
@@ -51,26 +75,85 @@ If not specified by a `*_unit` attribute, the [standard OpenMM unit system](http
-For other force fields with "library"-style charges, we could introduce a new section providing specific charges via SMIRKS pattern, though this is not yet implemented. Bond charge corrections, however, are supported as we discuss below.
+**QUESTION:** Since we're defining potentials instead of forces, do we want to remove the term `Force` from these tags and instead use the simpler `<Bonds/>`, `<Angles/>`, `<Torsions/>`, etc?
+
+## Charge models
+
+SMIRNOFF supports several ways for specifying electrostatic models.
+Currently, only classical fixed point charge models are supported, but extensions to the specification will support point multipoles, point polarizable dipoles, Drude oscillators, charge equilibration methods, and so on.
+
+### Library charges
+
+A mechanism is provided for specifying library charges should be used for molecules or residues that match library charge templates.
+
+For example, to assign partial charges
+```
+<LibraryCharges charge_unit="elementary_charge">
+   <!-- match a non-terminal alanine residue -->
+   <Residue name="ALA" smirks="[$([NX3H:1](C)(C)[H:2])][CX4H:3]([CH3X4:4]([H:5][H:6][H:7]))[CX3:8](=[OX1:9])([N])" charge1="-0.4157" charge2="0.2719" charge3="0.0337" charge4="-0.1825" charge5="0.0603" charge6="0.0603" charge7="0.0603" charge8="0.5973" charge9="-0.5679">
+   ...
+</LibraryCharges>
+```
+In this case, a SMIRKS string defining the residue tags each atom that should receive a partial charge, with the charges specified by `charge#` attributes.
+Note that the first match for a given set of atoms is used, so chemically equivalent atoms should be assigned the same charge to avoid undefined behavior.
+
+Solvent models or excipients can also have partial charges specified in this manner:
+```
+<LibraryCharges charge_unit="elementary_charge">
+   <!-- TIP3P water oxygen with charge override -->
+   <!-- WARNING: charges are not supported yet (see issue openforcefield#25), and they are currently here only for documentation -->
+   <Atom smirks="[#1:1]-[#8X2H2+0:2]-[#1:3]" charge1="+0.417" charge2="-0.834" charge3="+0.417"/>
+</LibraryCharges>
+```
+
+**QUESTION:** Is this an OK way to handle chemically equivalent atoms, like hydrogens? Is there a simpler way to avoid having to specify multiple copies of chemically equivalent atoms?
```

Comment:
This is how I would do it. I don't see a better way to handle it at this point. Someone could deliberately (or accidentally) use this to make it so make it so chemically equivalent atoms end up with different charges, but I don't see why we need to change the FORMAT to disallow that. Also, there are actually some reasons one might want to do this sometimes (Chris Fennell has been doing some work along these lines that is perhaps odd but is at least well motivated). If anything this is something I'd take care of at the "error/warning" level by eventually adding checking of symmetry.

Reply:
I think this is the best way to do it right now. Ultimately we may want to distinguish diastereotopic atoms: adjacent to a stereocenter. e.g. the two H's on the CH2 in CC[C@H](Cl)c1ccccc1 but this is usually a negligible effect.